### PR TITLE
Move custom templates code to the `CodeGenerator` class

### PR DESCRIPTION
**Pull request by pnbabu**
Move templates code from `NestCodeGenerator` to its parent `CodeGenerator` to make the code reusable for all targets.

### CODE REVIEW

**Comment by pnbabu** [comment]
> This looks great! Two minor questions: why the removed test, and there are a lot of unrelated formatting changes due to line length; are they causing our CI static check to error out? It should be ignoring line length (E501).

The test was a duplicate, hence removed. My IDE auto-formatted some lines, have undone them now.

File: pynestml/utils/with_options.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 
 class WithOptions:
-    r"""This class allows options (indexed by strings) to be set and retrieved. A set of default options may be supplied by overriding the `_default_options` member in the inheriting class."""
+    r"""This class allows options (indexed by strings) to be set and retrieved. A set of default options may be supplied by overriding the `_default_options` member in the inheriting class. """
```
Comment (clinssen) [inline pynestml/utils/with_options.py:28]: undo

File: pynestml/utils/with_options.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def option_exists(self, k: str) -> bool:
         return k in self._options.keys()
 
     def set_options(self, options: Mapping[str, Any]) -> Mapping[str, Any]:
-        r"""Set options. "Eats off" any options that it knows how to set, and returns the rest as "unhandled" options."""
+        r"""Set options. "Eats off" any options that it knows how to set, and returns the rest as "unhandled" options. """
```
Comment (clinssen) [inline pynestml/utils/with_options.py:47]: undo

**Review by clinssen — CHANGES_REQUESTED** [review]

**Comment by clinssen** [comment]
Many thanks!

**Pull request merged**